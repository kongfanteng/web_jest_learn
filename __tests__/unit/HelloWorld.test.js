import HelloWorld from "@/components/HelloWorld.vue";
import { shallowMount } from "@vue/test-utils";
import { createApp, h } from "vue";
describe("HelloWorld.vue", () => {
  it("renders props.msg when passed", () => {
    const msg = "HelloWorld";
    const wrapper = shallowMount(HelloWorld, {
      props: { msg },
    });
    // expect(wrapper.text()).toMatch(msg);
    // expect(wrapper.findAll(".hello").length).toBe(1);
    // expect(wrapper.props("msg")).toEqual("HelloWorld");
    expect(wrapper.element).toMatchSnapshot();
  });
});
