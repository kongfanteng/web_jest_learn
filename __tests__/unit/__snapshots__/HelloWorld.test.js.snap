// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld.vue renders props.msg when passed 1`] = `
<div
  class="hello"
>
  <h1>
    HelloWorld
  </h1>
</div>
`;
